internal/sharedcomponent/sharedcomponent.go

```diff
@@ -108,19 +108,30 @@ var _ componentstatus.Reporter = (*hostWrapper)(nil)
 type hostWrapper struct {
 	host           component.Host
 	sources        []componentstatus.Reporter
-	previousEvents []*componentstatus.Event
+	previousEvents *ring.Ring
 	lock           sync.Mutex
 }
 
 func (h *hostWrapper) GetExtensions() map[component.ID]component.Component {
 	return h.host.GetExtensions()
 }
 
+func (h *hostWrapper) hasEvent(e *componentstatus.Event) bool {
+	hasEvent := false
+	h.previousEvents.Do(func(a any) {
+		if previousEvent, ok := a.(*componentstatus.Event); ok && previousEvent == e {
```

Comment:
How can this happen?

Reply:
Added a test.  It can happen if a component were to ever create an event and then emit it twice.  Components should really never do that, but this defensive code makes sure that if they do we don't fill up our buffer with duplicate events.

Reply:
Another thing is that events have a "timestamp" right now witch makes them to be unique every time you create a new event, but there is a discussion to remove that, which will cause lots of "duplicate" events to be detected and that is wrong most likely.

So overall I think this is a "premature defensive code" (optimization) that we should remove.

Reply:
ok